### PR TITLE
Add the packaging metadata to build the guetzli snap

### PR DESCRIPTION
**Pull request by elopio**
This is a package for the secure installation of apps that works in most Linux distributions.
Landing it upstream will enable builds that then can be distributed to many users through the Ubuntu store.

This PR requires #15.

### CODE REVIEW

**Comment by elopio** [comment]
You can find more information about snapcraft here: http://snapcraft.io/

To test it in an Ubuntu 16.04 machine:

    $ sudo apt install git snapcraft
    $ git clone https://github.com/elopio/guetzli
    $ cd guetzli
    $ git checkout snapcraft
    $ snapcraft
    $ sudo snap install *.snap --dangerous
    
If you have any questions or comments, please let me know.

File: snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  equivalent quality generated by libjpeg. Guetzli generates only sequential
+  (nonprogressive) JPEGs due to faster decompression speeds they offer.
+
+grade: devel
+confinement: strict
+
+apps:
+  guetzli:
+    command: bin/Release/guetzli
+    plugs: [home]
```
Comment (robryk) [inline snapcraft.yaml]: Just curious: is there no way to say "this application has to access user-chosen files" and:
a) for commandline apps, consider paths mentioned on the commandline to be user-chosen
b) for GUI apps, require the app to use an external filepicker and consider files chosen in the filepicker as user-chosen?

It seems to kinda defeat the whole point of sandboxing if an app can modify ~/.bashrc

**Comment by robryk** [comment]
Thanks. I've merged this manually.

I don't have Ubuntu 16 on hand (only Ubuntu 14), so I'd appreciate if you checked that this actually works.

**Comment by robryk** [comment]
@elopio How does this work with 32bit and 64bit systems? When you compile Guetzli with make, it will build for whatever architecture $CXX will by default build for. I am under the impression that precompiled snaps do get distributed.

File: snapcraft.yaml (hunk 1 above)
Comment (elopio) [inline snapcraft.yaml]: The home interface excludes dot files, so the user can't read or write to .bashrc.

This is a transitional interface, which offers a little more access than what we would like, to ease the transition to snaps. In the ubuntu phone we have that filepicker, it's called media hub and some of my teammates are working on snap support there. For CLI applications, we are first collecting feedback from the use of the home and removable-media interfaces, and the classic-confinement, before settling in a final design for file access out of the safe snap space. 

So any feedback you can give us here will be highly appreciated.

**Comment by elopio** [comment]
Hello @robryk. Thanks for merging, and sorry for the late reply. I was on holidays, but now I'm back, and happy to help you publishing the snap if you need a hand :)

The snap will include the precompiled binaries. Our preferred solution for building snaps for multiple architectures is the build farm in launchpad. You give it a git repo with a snapcraft.yaml and it will build and publish one snap for each supported architecture. http://snapcraft.io/docs/build-snaps/ci-integration#using-launchpad

In our opinion, that simplifies the problem a lot because you don't have to take care of multiple toolchains in travis or jenkins. However, we are talking about supporting builds for other architectures for cases like go, that make it really simple and clean. And you can always build the snap as part of your custom multi-arch build, like https://github.com/syncthing/syncthing/blob/master/build.go#L566

Again, we would appreciate your comments about different architectures to help us assign priorities in our backlog.

¡pura vida!

**Pull request closed without merging**